fmdtools/analyze/graph.py

```diff
@@ -1559,48 +1878,124 @@ class ASGGraph(Graph):
     def __init__(self, asg, withstates=True):
         self.g = nx.compose(asg.flow_graph, asg.action_graph) 
         self.set_nx_labels(asg)
-        if withstates: self.set_nx_states(asg)
+        if withstates:
+            self.set_nx_states(asg)
+
     def set_nx_labels(self, asg):
+        """
+        TODO : requires documentation
```

Comment:
Labels the underlying networkx graph structure with type attributes corresponding to the ASG.

asg : ASG
    Action Sequence Graph object to represent

Reply:
Done